frontend/src/ui/Blocks.tsx

```diff
@@ -57,12 +57,77 @@ const TextBlock: React.FC<TextProps> = ({ title, content }) => (
     </Block>
 );
 
-const VideoListBlock: React.FC<{ title: string | null }> = ({ title }) => (
-    <Block>
-        <Title title={title} />
-        <i>not yet implemented ☹️</i>
-    </Block>
-);
+const VideoListBlock: React.FC<{ title: string | null }> = ({ title }) => {
+    const [THUMB_WIDTH, THUMB_HEIGHT] = [16, 9].map(x => x * 13);
+
+    return (
+        <Block>
+            <Title title={title} />
+            <div css={{
+                display: "flex",
+                flexWrap: "wrap",
+            }}>
+                {DUMMY_VIDEOS.map(v => {
+                    const url = `/v/${v.id}`;
+
+                    return (
+                        <div
+                            key={v.id}
+                            css={{
+                                margin: "12px 8px",
+                                width: THUMB_WIDTH,
+                                "& a": { color: "black", textDecoration: "none" },
+                            }}
+                        >
+                            <a href={url} css={{ position: "relative", display: "block" }}>
+                                <img
+                                    src={v.thumbnail}
+                                    width={THUMB_WIDTH}
+                                    height={THUMB_HEIGHT}
+                                    css={{ display: "block" }}
+                                />
+                                <div css={{
+                                    position: "absolute",
+                                    right: 6,
+                                    bottom: 6,
+                                    backgroundColor: "hsla(0, 0%, 0%, 0.75)",
+                                    border: "1px solid black",
+                                    borderRadius: 4,
+                                    padding: "0 4px",
+                                    color: "white",
+                                }}>
+                                    {formatLength(v.length)}
+                                </div>
+                            </a>
+
+                            <h3 css={{
+                                fontSize: 16,
+                            }}>
+                                <a href={url}>{v.title}</a>
+                            </h3>
+                        </div>
+                    );
+                })}
+            </div>
+        </Block>
+    );
+};
+
+const formatLength = (totalSeconds: number) => {
+    const seconds = totalSeconds % 60;
+    const minutes = Math.floor(totalSeconds / 60) % 60;
+    const hours = Math.floor(totalSeconds / (60 * 60));
+
+    const pad = (n: number) => (n < 10 ? `0${n}` : `${n}`);
+
+    if (hours > 0) {
+        return `${hours}:${pad(minutes)}:${pad(seconds)}`;
+    } else if (minutes > 0) {
+        return `${minutes}:${pad(seconds)}`;
+    } else {
+        return `0:${pad(seconds)}`;
```

Comment:
```suggestion
    } else {
        return `${minutes}:${pad(seconds)}`;
```

Shouldn't this be the same (ignoring `minutes < 0`, which should not happen)? :thinking:

Reply:
Ha, good point :D